src/System.Net.Http/src/System/Net/Http/Headers/HttpResponseHeaders.cs

```diff
@@ -141,9 +143,11 @@ public HttpHeaderValueCollection<WarningHeaderValue> Warning
 
         #endregion
 
-        internal HttpResponseHeaders()
-            : base(HttpHeaderType.General | HttpHeaderType.Response | HttpHeaderType.Custom, HttpHeaderType.Request)
+        internal HttpResponseHeaders(bool containsTrailingHeaders = false)
+            : base(containsTrailingHeaders ? HttpHeaderType.All ^ HttpHeaderType.Request : HttpHeaderType.General | HttpHeaderType.Response | HttpHeaderType.Custom,
```

Comment:
`All` includes `NonTrailing`, but is used when `contrainsTrailingHeaders = true`.

Reply:
It doesn't matter in this case because all mask based header type checks must also be changed to accommodate for two variations (with NonTrailing flag and without) of the given header type value under the consideration. See my comment below.